classes/item.rb

```diff
@@ -0,0 +1,45 @@
+require 'date'
+class Item
+  attr_accessor :id, :genre, :author, :label, :publish_date
+
+  def initialize(params = {})
+    @id = params[:id] || Random.rand(1..1000)
+    @genre = params[:genre]
+    @author = params[:author]
+    @label = params[:label]
+    @source = params[:source]
+    @publish_date = (Date.strptime(params[:publish_date], '%Y-%m-%d') if params[:publish_date])
+    @archived = false
+  end
+
+  def move_to_archive
+    return unless can_be_archived?
+
+    @archived = true
+  end
+
+  def add_label=(label)
+    @label = label
+    label.items.push(self) unless label.items.include?(self)
+  end
+
+  def add_author=(author)
+    @author = author
+    author.items.push(self) unless author.items.include?(self)
+  end
+
+  def add_genre=(genre)
+    @genre = genre
+    genre.items.push(self) unless genre.items.include?(self)
+  end
+
```

Comment:
- [ ] Well done working on your item class however I recommend adding the source as you are a group of four:


`def source=(source)
      @source = source
      source.items.push(self) unless source.items.include?(self)
    end
`

Reply:
well resolved! ✔️